### PR TITLE
Fix string compare, java detection, version string

### PR DESCRIPTION
**Pull request by mrallen1**
Fixing a few issues:

* I forgot to set a new version string on the past two releases. Deeeeerrrpp. (Sorry)
* On macOS when you execute `/usr/bin/java{c}` on a system without Java installed it pops up this super annoying dialog box.  The OTP build does it too, but we can avoid an extra one in kerl by probing `/usr/libexec/java_home` 
* When comparing strings, bash wants the `!=` operator, not the integer comparison `-ne`. Sorry.

### CODE REVIEW

**Comment by sanmiguel** [comment]
Sorry for the CI noise - hoping to get back to getting Travis running over the weekend. Right now it will fail because there's no configuration outside of #183 but it's enabled everywhere. Can temporarily disable it if it's getting annoying!

**Comment by mrallen1** [comment]
Ehn, it's fine. Think these would be good to go in sooner than later, so if you get a moment, can you review ?

File: kerl [hunk 1]
```diff
@@ -411,8 +411,10 @@ assert_perl()
 
 get_javac_version()
 {
-    java_loc=$(which javac)
-    if [ -z "$java_loc" ]; then
+    # http://apple.stackexchange.com/a/135079
+    /usr/libexec/java_home >/dev/null 2>&1
```
Comment (sanmiguel) [inline kerl]: Does this hold up on other systems? It doesn't look like it works on my Ubuntu 14.04 server, at least - or is this function only called on MacOS?

File: kerl [hunk 2]
```diff
@@ -589,7 +591,7 @@ _do_build()
     if [ -e "./$KERL_CONFIG_STORAGE_FILENAME.md5" ]; then
         # Compare our current options to the saved ones
         OLD_SUM=$(read -r < "./$KERL_CONFIG_STORAGE_FILENAME.md5")
-        if [ "$SUM" -ne "$OLD_SUM" ]; then
+        if [ "$SUM" != "$OLD_SUM" ]; then
```
Comment (sanmiguel) [inline kerl]: Good catch!

File: kerl [hunk 3]
```diff
@@ -24,7 +24,7 @@
 
 unset ERL_TOP
 
-KERL_VERSION="1.3.4"
+KERL_VERSION="1.4.1"
```
Comment (sanmiguel) [inline kerl]: Have you had any thoughts on #188 ?

File: kerl (hunk 1 above)
Comment (mrallen1) [inline kerl]: Does that file not exist when you install OpenJDK on a Linux box? Hmm. Maybe for Linux we just directly check to see if javac exists...

File: kerl (hunk 3 above)
Comment (sanmiguel) [inline kerl]: Also let's not forget to bump this to `1.4.2` before creating the next release :)

File: kerl (hunk 1 above)
Comment (sanmiguel) [inline kerl]: Can we roll this java-related change into a separate PR? Then we can go ahead and merge the rest of this and figure out the java stuff without blocking it?

Comment (mrallen1) [inline kerl]: Yeah, good idea.

File: kerl (hunk 3 above)
Comment (mrallen1) [inline kerl]: 👍

**Comment by mrallen1** [comment]
Closing in favor of #191

**Pull request closed without merging**